File: src/Prob17LetterCombinations/SolutionTest.java
package Prob17LetterCombinations;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.List;
import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.*;

class SolutionTest {

    public static Stream<Arguments> leetCode() {
        return Stream.of(
                Arguments.of("23", List.of(
                        "ad","ae","af","bd","be","bf","cd","ce","cf"
                )),
                Arguments.of("", List.of()),
                Arguments.of("2", List.of("a", "b", "c"))
        );
    }

    @ParameterizedTest
    @MethodSource("leetCode")
    void letterCombinations(String x, List<String> expected) {
        assertEquals(expected, new Solution().letterCombinations(x));
    }
}

File: tests/Prob1688/SolutionTest.kt
package Prob1688

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.params.ParameterizedTest
import org.junit.jupiter.params.provider.Arguments
import org.junit.jupiter.params.provider.MethodSource
import java.util.stream.Stream

internal class SolutionTest {
    @ParameterizedTest
    @MethodSource("leetCodeSource")
    fun leetCode(input: String, word: String, expected: Int){
        assertEquals(expected, Solution().maxRepeating(input, word));
    }

    companion object {
        @JvmStatic
        fun leetCodeSource(): Stream<Arguments> {
            return Stream.of(
                Arguments.of("ababc", "ab", 2),
                Arguments.of("ababc", "ba", 1),
                Arguments.of("ababc", "ac", 0),
                Arguments.of("ab", "ab", 1),
                Arguments.of("aaabaaaabaaabaaaabaaaabaaaabaaaaba", "aaaba", 5)
            )
        }
    }


}